util/src/main/scala/org/ensime/util/Debouncer.scala

```diff
@@ -0,0 +1,118 @@
+// Copyright: 2010 - 2017 https://github.com/ensime/ensime-server/graphs
+// License: http://www.gnu.org/licenses/gpl-3.0.en.html
+package org.ensime.util
+
+import akka.actor.Scheduler
+import java.time.Clock
+import java.util.concurrent.atomic.{ AtomicBoolean, AtomicInteger }
+import org.slf4j.LoggerFactory
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration._
+
+/**
+ * Generic debouncer implementation
```

Comment:
I explained in another ticket. The primary reason is performance. Debounced calls are filtered instantaneously with almost zero allocations (except the stack frame allocation). This is meaningful when you consider that I have seen roughly 20,000 invocations / second on average filtered by this logic.

There are a lot of allocations and mutations involved with using an actor.

Also, this is typesafe.

Let me switch it: why use an Actor?